client/dashboard/sites/settings-site-visibility/privacy-form.tsx

```diff
@@ -0,0 +1,144 @@
+import { DataForm } from '@automattic/dataviews';
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	Button,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState } from 'react';
+import type { SiteSettings } from '../../data/types';
+import type { Field } from '@automattic/dataviews';
+import type { UseMutationResult } from '@tanstack/react-query';
+
+interface SitePrivacy {
+	visibility: string;
+}
+
+const fields: Field< SitePrivacy >[] = [
+	{
+		id: 'visibility',
+		Edit: ( { field, onChange, data } ) => {
+			const { id, getValue } = field;
+			const value = getValue( { item: data } );
+
+			let help;
+			if ( value === 'coming-soon' ) {
+				help = __(
+					'Your site is hidden from visitors behind a "Coming Soon" notice until it is ready for viewing.'
+				);
+			} else if ( value === 'public' ) {
+				help = __( 'Your site is visible to everyone.' );
+			} else {
+				help = __(
+					'Your site is only visible to you and logged-in members you approve. Everyone else will see a log in screen.'
+				);
+			}
+
+			return (
+				<ToggleGroupControl
+					__next40pxDefaultSize
+					__nextHasNoMarginBottom
+					isBlock
+					label=""
+					help={ help }
+					value={ value }
+					onChange={ ( newValue ) => {
+						onChange( { [ id ]: newValue } );
+					} }
+				>
+					<ToggleGroupControlOption label="Coming soon" value="coming-soon" />
```

Comment:
We need to localize the labels.

Reply:
Thanks, somehow I totally missed this... updated.